### ISSUE TITLE
4.1.1 events: failed to read protocol version

### ISSUE THREAD

**Issue by cubed-it** [CrunchyData/postgres-operator#1119]
I'm running 4.1.1 on AKS 1.14.8 and I observe constant Liveness/Readiness probe failed. Is this known?

events log:
```
[nsqd] 2019/12/05 14:59:22.899954 INFO: TCP: new client(10.244.0.1:54834)
[nsqd] 2019/12/05 14:59:22.900011 ERROR: failed to read protocol version - EOF
[nsqd] 2019/12/05 14:59:24.598864 INFO: TCP: new client(10.244.0.1:54888)
[nsqd] 2019/12/05 14:59:24.598960 ERROR: failed to read protocol version - EOF
[nsqd] 2019/12/05 14:59:29.599034 INFO: TCP: new client(10.244.0.1:55010)
[nsqd] 2019/12/05 14:59:29.599114 ERROR: failed to read protocol version - EOF
[nsqd] 2019/12/05 14:59:32.899860 INFO: TCP: new client(10.244.0.1:55070)
[nsqd] 2019/12/05 14:59:32.899883 ERROR: failed to read protocol version - EOF
[nsqd] 2019/12/05 14:59:34.598817 INFO: TCP: new client(10.244.0.1:55110)
[nsqd] 2019/12/05 14:59:34.599023 ERROR: failed to read protocol version - EOF
[nsqd] 2019/12/05 14:59:44.598897 INFO: TCP: new client(10.244.0.1:55334)
[nsqd] 2019/12/05 14:59:44.598921 ERROR: failed to read protocol version - EOF
Signal trap triggered, beginning shutdown..
++ trap_sigterm
++ echo 'Signal trap triggered, beginning shutdown..'
+++ pidof nsqd
/usr/local/bin/pgo-event.sh: line 18: pidof: command not found
++ kill -9
kill: usage: kill [-s sigspec | -n signum | -sigspec] pid | jobspec ... or kill -l [sigspec]
+++ pidof nsqadmin
/usr/local/bin/pgo-event.sh: line 19: pidof: command not found
++ kill -9
kill: usage: kill [-s sigspec | -n signum | -sigspec] pid | jobspec ... or kill -l [sigspec]
end of pgo-event
+ echo 'end of pgo-event'
```

To mitigate this issue I modified the event deployment. Is this okay?
```
            "livenessProbe": {
              "httpGet": {
                "path": "/ping",
                "port": 4151,
                "scheme": "HTTP"
              },
              "initialDelaySeconds": 15,
              "timeoutSeconds": 1,
              "periodSeconds": 5,
              "successThreshold": 1,
              "failureThreshold": 3
            },
            "readinessProbe": {
              "httpGet": {
                "path": "/ping",
                "port": 4151,
                "scheme": "HTTP"
              },
              "initialDelaySeconds": 15,
              "timeoutSeconds": 1,
              "periodSeconds": 5,
              "successThreshold": 1,
              "failureThreshold": 3
            },
```

Also is it okay that apiserver is spamming the following error? However it does not restart...
```
2019/12/05 15:38:30 http: TLS handshake error from 10.244.0.1:35954: EOF
```

**Comment by wilybrace** [comment]
> I'm running 4.1.1 on AKS 1.14.8 and I observe constant Liveness/Readiness probe failed. Is this known?

The error messages are known and being addressed as part of the same fix as #1111
To be explicit, the checks are not failing but the servers (nsqd and the pgo-apiserver) do not have provisions for quietly handling a tcpsocket connection test.

**Comment by cubed-it** [comment]
nsqd tcp liveness/readiness probe definitely fail for me. I can't get to the log right now, but k8s tells me this clearly when I describe the pod. apiserver is indeed fine.

**Comment by jkatz** [comment]
This should be addressed in #1122 which will be available in the 4.1.2 release.

(It is also addressed in #1121 which will be available in the 4.2.0 release).

Thanks for reporting the issue!
